docs/developer_guide/testing_suite.rst

```diff
@@ -0,0 +1,63 @@
+Running GIN tests locally
+=========================
+
+NeuroConv verifies the integrity of all code changes by running a full test suite on short examples of real data from the formats we support.
+
+The testing suite is broken up into sub-folders based on the scope of functionalities and dependencies you wish to test.
+
+We recommend always running tests in a fresh environment to ensure errors are not the result of contaminated dependencies.
+
+There are three broad classes of tests in this regard...
+
+
+
+Minimal
+-------
+
+These test internal functionality using only minimal dependencies or pre-downloaded data.
+
+Subfolders: `tests/test_minimal <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_minimal>`_ and `tests/test_internals <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_internals>`_
+
+These can be run using only ``pip install -e neuroconv[test]`` and calling ``pytest tests/test_minimal`` or ``pytest tests/test_internal``.
+
+
+
+Modality
+--------
+
+These test the functionality of our write tools tailored to certain external dependencies.
+
+Subfolders: `tests/test_ophys <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_ophys>`_, `tests/test_ecephys <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_ecephys>`_, `tests/test_behavior <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_behavior>`_, and `tests/test_text <https://github.com/catalystneuro/neuroconv/tree/main/tests/test_text>`_
+
+These can be run in isolation using ``pip install -e neuroconv[test,<modality>]`` and calling ``pytest tests/test_<modality>`` where ``<modality>`` can be any of ``ophys``, ``ecephys``, ``text``, or ``behavior``.
+
+
+
+On Data
+-------
+
+For proprietary formats, we like to be able to regularly test our conversions against small snnippets of real data.
+
+To reduce the complexity of the testing suite, the tests simply assume that the example datasets exist somewhere on your system.
+
+These can each by downloaded using `Datalad <https://www.datalad.org/>`_ and the example datasets can be found at
+
+For electrophysiology data
+""""""""""""""""""""""""""
+``datalad install -rg https://gin.g-node.org/NeuralEnsemble/ephy_testing_data``
+
+For optical physiology data
+"""""""""""""""""""""""""""
+``datalad install -rg https://gin.g-node.org/CatalystNeuro/ophys_testing_data``
+
+For behavioral data
+"""""""""""""""""""
+``datalad install -rg https://gin.g-node.org/CatalystNeuro/behavior_testing_data``
```

Comment:
how do you update the data in these folders?

Reply:
I think `datalad update`